spec/support/compare_locales_support.rb

```diff
@@ -0,0 +1,36 @@
+module CompareLocalesSupport
+
+  LOCALE_1 = './config/locales/en.yml'
+  LOCALE_2 = './config/locales/es.yml'
+
+  require 'yaml'
+
+  def self.flatten_keys(hash, prefix='')
+    hash.keys.each_with_object([]) do |key, keys|
+      if hash[key].respond_to?(:keys)
+        keys.concat(flatten_keys(hash[key], "#{prefix}#{key}."))
+      else
+        keys << "#{prefix}#{key}"
+      end
+    end
+  end
+
+  def self.compare_locale_hashes(primary_locale, locale_to_compare)
+    primary_keys = flatten_keys(primary_locale[primary_locale.keys.first])
+    keys_to_compare = flatten_keys(locale_to_compare[locale_to_compare.keys.first])
```

Comment:
I'm not terribly concerned about the name—although `comparison_locale` would add a little consistency—but it's easier to read as `primary_locale.values.first` rather than `primary_locale[primary_locale.keys.first]`